src/workerd/api/web-socket.c++

```diff
@@ -577,7 +576,6 @@ void WebSocket::ensurePumping(jsg::Lock& js) {
     //   refactoring.
     context.awaitIoLegacy(kj::mv(promise)).then(js, [this, thisHandle = JSG_THIS](jsg::Lock& js) {
```

Comment:
As mentioned in the chat, I'm not convinced this really needs to use `awaitIoLegacy` here. If you use something like...

```
ioContext.addTask(promise.then(js,
    ioContext.addFunctor([this, thisHandle=JSG_THIS)() {
  auto& native = *farNative;
  if (native.outgoingAborted) {
    return IoContext::current().run([](jsg::Lock& js) { reportError(...) });
  }
  // ...
  return kj::READY_NOW;
});
```

I think it'll simplify the timing just a bit. Bringing a JS promise into the mix seems unnecessary.

Reply:
If `promise` is a KJ promise, its `.then()` won't run under the isolate lock, so that code won't work. That's what `awaitIo()` (or `awaitIoLegacy()`) is needed for.